src/submission/submission-to-clinical/stat-calculator.ts

```diff
@@ -76,141 +96,125 @@ const getEmptyCoreStats = (): CompletionStats => ({
 });
 
 /**
- * This is the main core stat calculation function.
- * We consider only `required & core` fields for core field calculation, which are always submitted.
+ * Main calculation logic for core-completion state for the donor.
  *
- * Note: This function is referenced in the recalculate-core-completion migration file.
- * Any code updates should validate that migration is unaffected.
+ * If the donor has an exception allowing some core entities to be missing, that must be provided in the function arguments.
+ * This will excuse the donor from requiring a treatment or followup entity to be core complete.
+ *
+ * A donor that is marked core-complete (coreCompletionPercentage === 1) cannot lose the complete status.
+ * The individual core entities will still be calculated so the current list of complete entities for this donor will still be known.
  *
  * @param donor
- * @param forceFlags
+ * @param hasMissingEntityException - Indicates if this donor has been given an exception to allow it to be core complete while missing select core entities
  * @returns
  */
-export const calcDonorCoreEntityStats = (
-	donor: Donor | DeepReadonly<Donor>,
-	forceFlags: ForceRecaculateFlags, // used to control recalculate under certain conditions
-) => {
-	// Intentionally create a mutable Donor copy for coreCompletion updates
-	const donorCopy = cloneDeep(donor) as Donor;
-
-	let newCompletionStats = donorCopy.completionStats
-		? { ...donorCopy.completionStats }
+export const calculateDonorCoreCompletionStats = (
+	donor: Donor,
+	hasMissingEntityException: boolean,
+): CompletionStats => {
+	let newCompletionStats: CompletionStats = donor.completionStats
```

Comment:
this doesn't need to `let`
`const` is memory of object itself, not properties of object

Reply:
right you are.